### ISSUE TITLE
Websocket fast refresh doesn't work with assetPrefix

### ISSUE THREAD

**Issue by genetschneider** [vercel/next.js#30328]
### What version of Next.js are you using?

12.0.0

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

Defining `assetPrefix` in next.config.js will result in local fast refresh failing: 
```
Uncaught DOMException: Failed to construct 'WebSocket': The URL 'ws://localhost:3000custom-asset-prefix/_next/webpack-hmr' is invalid.
    at init (webpack-internal:///./node_modules/next/dist/client/dev/error-overlay/websocket.js:42:14)
    at Object.connectHMR (webpack-internal:///./node_modules/next/dist/client/dev/error-overlay/websocket.js:28:3)
    at eval (webpack-internal:///./node_modules/next/dist/client/next-dev.js:35:17)
    at Object../node_modules/next/dist/client/next-dev.js (http://localhost:3000/custom-asset-prefix/_next/static/chunks/main.js?ts=1635267083870:828:1)
    at Object.options.factory (http://localhost:3000/custom-asset-prefix/_next/static/chunks/webpack.js?ts=1635267083870:685:31)
    at __webpack_require__ (http://localhost:3000/custom-asset-prefix/_next/static/chunks/webpack.js?ts=1635267083870:37:33)
    at __webpack_exec__ (http://localhost:3000/custom-asset-prefix/_next/static/chunks/main.js?ts=1635267083870:1430:61)
    at http://localhost:3000/custom-asset-prefix/_next/static/chunks/main.js?ts=1635267083870:1431:37
    at webpackJsonpCallback (http://localhost:3000/custom-asset-prefix/_next/static/chunks/webpack.js?ts=1635267083870:1268:39)
    at http://localhost:3000/custom-asset-prefix/_next/static/chunks/main.js?ts=1635267083870:9:61
```

### Expected Behavior

`assetPrefix` can be defined locally. `assetPrefix` can be a simple string or a url (both locally and in production). Local fast refresh works with custom `assetPrefix`. 

### To Reproduce

Fully reproducible repository can be found [here](https://github.com/genetschneider/nextjs-websocket-fast-refresh-bug)

**Comment by aviculturist** [comment]
defining an `assetPrefix` is critical to enable deploys to IPFS, so this is a blocking bug for me as well.

**Comment by timneutkens** [comment]
Fixed in #30632

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
